plugins/webapi/info/plugin.go

```diff
@@ -0,0 +1,92 @@
+package info
+
+import (
+	"net/http"
+
+	"github.com/iotaledger/goshimmer/plugins/autopeering/local"
+	"github.com/iotaledger/goshimmer/plugins/banner"
+	"github.com/iotaledger/goshimmer/plugins/webapi"
+	"github.com/iotaledger/hive.go/node"
+	"github.com/labstack/echo"
+	"github.com/mr-tron/base58"
+)
+
+var PLUGIN = node.NewPlugin("WebAPI info Endpoint", node.Enabled, configure)
+
+func configure(plugin *node.Plugin) {
+	webapi.Server.GET("info", getInfo)
+}
+
+// getInfo returns the info of the node
+// e.g.,
+// {
+// 	"version":"v0.2.0",
+// 	"identity":"7BxV1v3nFHefn4J88jeZebqnJRvSHt1jC7ME6tmKLhy7",
+// 	"publickey":"CjUsn86jpFHWnSCx3NhWfU4Lk16mDdy1Hr7ERSTv3xn9",
+// 	"enabledplugins":[
+// 		"Config",
+// 		"Autopeering",
+// 		"Analysis",
+// 		"WebAPI data Endpoint",
+// 		"WebAPI dRNG Endpoint",
+// 		"MessageLayer",
+// 		"CLI",
+// 		"Database",
+// 		"DRNG",
+// 		"WebAPI autopeering Endpoint",
+// 		"Metrics",
+// 		"PortCheck",
+// 		"SPA",
+// 		"WebAPI",
+// 		"WebAPI info Endpoint",
+// 		"WebAPI message Endpoint",
+// 		"Banner",
+// 		"Gossip",
+// 		"Graceful Shutdown",
+// 		"Logger"
+// 	],
+// 	"disabledlugins":[
+// 		"Graph",
+// 		"RemoteLog",
+// 		"Spammer",
+// 		"WebAPI Auth"
+// 	]
+// }
+func getInfo(c echo.Context) error {
+	enabledPlugins := []string{}
+	disabledPlugins := []string{}
+	for plugin, status := range node.GetPlugins() {
+		switch status {
+		case node.Disabled:
+			disabledPlugins = append(disabledPlugins, plugin)
+		case node.Enabled:
+			enabledPlugins = append(enabledPlugins, plugin)
+		default:
+			continue
+		}
+	}
+
+	return c.JSON(http.StatusOK, Response{
+		Version:         banner.AppVersion,
+		Identity:        base58.Encode(local.GetInstance().Identity.ID().Bytes()),
+		PublicKey:       base58.Encode(local.GetInstance().PublicKey().Bytes()),
```

Comment:
Use the `String()` methods here. Then you also don't need to worry about different representations :)

Reply:
The String() method of the Identity.ID() returns hex.EncodeToString(id[:8])
Should we use that?

Reply:
I think it's also about consistency here. If we use the method everywhere it's easy to change the representation just at one place. I don't mind it being the full ID or just part of it, I just want it to be consistent! :)

Reply:
I meant, my point is that if we want to use the String method everywhere we need to change its implementation (from hex encoded to base58 encoded) and I did not want to touch the autopeering.

Reply:
Unless of course you are fine with hex encoded for the identity